emcc.py

```diff
@@ -1242,6 +1242,12 @@ def check(input_file):
       if shared.Settings.WASM:
         shared.Settings.BINARYEN = 1 # these are synonyms
 
+        # When only targeting wasm, the .asm.js file is not executable, so is treated as an intermediate build file that can be cleaned up.
+        if shared.Building.is_wasm_only():
+          asm_target = asm_target.replace('.asm.js', '.temp.asm.js')
```

Comment:
If we do not rename, this would overwrite an existing `.asm.js` file that the user might have previously had from an earlier compilation. The idea being that since `-s BINARYEN_METHOD="'native-wasm,asmjs'"` is not that ideal, we want to support the sequence of two commands

    em++ a.cpp -o a.js --separate-asm
    em++ a.cpp -o a.js -s BINARYEN=1

which output to the same directory to generate a set of `.js`, `.asm.js`, `.js.mem` and .`wasm` files that one can use a custom shell file to create the compatibility fallback. If one doesn't rename the `.asm.js` file under a temp name, then the second build would overwrite the `.asm.js` that was produced on the first build.